kolibri/core/auth/hooks.py

```diff
@@ -0,0 +1,29 @@
+from kolibri.plugins.hooks import define_hook
+from kolibri.plugins.hooks import KolibriHook
+
+
+@define_hook
+class FacilityDataSyncHook(KolibriHook):
```

Comment:
This could be separated out into a pre-hook and post-hook, which would then allow each method to be setup with an `@abc.abstractmethod` decorator, which then gives a clear requirement that the method be instantiated.